### PR TITLE
feat(test) Add acceptance test that is backed by snuba

### PR DESCRIPTION
**Pull request by markstory**
Add a `requires_snuba` pytest marker so that we can integrate snuba based tests into our acceptance tests and gracefully handle snuba not being available. I've refactored the other service availability checks to only make one connect call in an attempt to keep tests faster.

I've added only a single acceptance test that uses snuba, as I assume my `.travis.yml` changes will be full of 🐍 and require additional care.

Refs SEN-246

### CODE REVIEW

File: tests/snuba/api/endpoints/test_event_details.py [hunk 1]
```diff
@@ -7,12 +7,15 @@
 from django.core.urlresolvers import reverse
 
 from sentry.models import UserReport, Group
-from sentry.testutils import APITestCase, SnubaTestCase
+from sentry.testutils import APITestCase, SnubaTestMixin
+from sentry.testutils.skips import requires_snuba
 
 
-class EventDetailsTest(APITestCase, SnubaTestCase):
+@requires_snuba
```
Comment (wedamija) [inline tests/snuba/api/endpoints/test_event_details.py]: Do we need to have these decorators on tests that are in `tests/snuba`? Seems like it's already a requirement there, and we wouldn't want to skip them if snuba isn't available.

File: tests/snuba/api/endpoints/test_event_details.py [hunk 2]
```diff
@@ -7,12 +7,15 @@
...
+class EventDetailsTest(SnubaTestMixin, APITestCase):
     def setUp(self):
         super(EventDetailsTest, self).setUp()
+        self.init_snuba()
```
Comment (wedamija) [inline tests/snuba/api/endpoints/test_event_details.py]: I don't really like that we need to call this for every test using the mixin. Could we keep `SnubaTestCase` and just have it call `init_snuba` for us? Not sure there are any cases in the existing tests where we don't want this to happen

File: tests/snuba/api/endpoints/test_event_details.py (hunk 1 above)
Comment (markstory) [reply tests/snuba/api/endpoints/test_event_details.py]: I was hoping to merge the snuba directory in with the other tests. Having endpoint tests in two different places doesn't seem great long term. With this we could have snuba tests live with the other endpoint tests, and we could remove the separate build for snuba only tests.

File: tests/snuba/api/endpoints/test_event_details.py (hunk 2 above)
Comment (markstory) [reply tests/snuba/api/endpoints/test_event_details.py]: Its possible, I was bumping into issues with inheritance with `AcceptanceTestCase`. I can revisit that though.

**Review by wedamija — APPROVED** [review]

**Pull request merged**